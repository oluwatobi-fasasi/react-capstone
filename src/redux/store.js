import { configureStore } from '@reduxjs/toolkit';
import dragonSlice from './dragons/dragonsSlice';
import missionReducer from './mission/missionslice';
import rocketSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    dragon: dragonSlice,
    mission: missionReducer,
    rocket: rocketSlice,
  },
});

export default store;
