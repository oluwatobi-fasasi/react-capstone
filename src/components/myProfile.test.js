import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Profile from './myProfile';

it('Test if the Profile page renders dragon element correctly', () => {
  const Dragon = [];
  const renderedDragon = renderer.create(
    <Provider store={store}>
      <Profile Dragon={Dragon} />
    </Provider>,
  );

  expect(renderedDragon).toMatchSnapshot();
});
