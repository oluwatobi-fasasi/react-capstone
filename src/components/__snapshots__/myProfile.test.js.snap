// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the Profile page renders dragon element correctly 1`] = `
<div
  className=""
>
  <div
    className="prof-rock-div"
  >
    <h2>
      My Rockets:
    </h2>
    <ul
      className="prof-rock-ul"
    />
  </div>
  <div
    className="prof-rock-div"
  >
    <h2>
      My Dragon:
    </h2>
    <ul
      className="prof-rock-ul"
    />
  </div>
</div>
`;
